idea/src/org/jetbrains/kotlin/idea/intentions/ConvertTwoComparisonsToRangeCheckIntention.kt

```diff
@@ -124,21 +128,71 @@ class ConvertTwoComparisonsToRangeCheckIntention : SelfTargetingOffsetIndependen
         // To avoid possible side effects
         if (!min.isSimple() || !max.isSimple()) return null
 
+        val valContext = value.analyze()
```

Comment:
You don't need to call `analyze()` multiple times; calling it once will return a `BindingContext` with information about all the expressions.